lib/facility_factory.rb

```diff
@@ -0,0 +1,31 @@
+class Facility_Factory
+
+  def initialize
+  end
+
+  def create_office_locations(locations)
+    facilities = locations.map do |facility|
+      #CO
+      if facility[:dmv_office] != nil
+        Facility.new(
+          name: facility[:dmv_office],
+          address: "#{facility[:address_li]} #{facility[:address_l]} #{facility[:city]} #{facility[:state]} #{facility[:zip]}",
+          phone: facility[:phone]
+        )
+        #NY
+      elsif facility[:office_name] != nil
+        Facility.new(
+          name: facility[:office_name],
+          address: "#{facility[:street_address_line_1]} #{facility[:street_address_line_2]} #{facility[:city]} #{facility[:state]} #{facility[:zip_code]}",
+          phone: facility[:public_phone_number]
+        )
+      else facility[:name] != nil
+        Facility.new(
+          name: facility[:name],
+          address: "#{facility[:address1]} #{facility[:city]} #{facility[:state]} #{facility[:zipcode]}",
+          phone: facility[:phone]
+        )
+      end
```

Comment:
You could consider creating a helper method for each of these conditions, especially since the data parsing logic is a little bit different across all of them.
```
if facility[:dmv_office] != nil
  new_co_facility(facility)
elsif facility[:office_name] != nil
  new_ny_facility(facility)
else # note - unlike elsif, else does not take a condition
  new_mo_facility(facility)
end
```